src/matcha_ml/core/core.py

```diff
@@ -132,7 +152,17 @@ def provision(
     password: str,
```

Comment:
The docstring for this wants updating to something more generic, e.g., "the password for the deployment server" (or similar) - users are probably going to think "what's the ZenServer"